independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -96,7 +95,11 @@ public static boolean isResourcePresentAtRuntime(String resourcePath) {
     }
 
     private final String name;
-    private final List<ClassPathElement> elements;
+    // the ClassPathElements to consider are normalPriorityElements + lesserPriorityElements
+    private final List<ClassPathElement> normalPriorityElements;
+    private final List<ClassPathElement> lesserPriorityElements;
```

Comment:
As explained in https://github.com/quarkusio/quarkus/pull/42525/files#r1719580032, I changed the semantic a bit here.